rs/registry/canister/src/get_node_providers_monthly_xdr_rewards.rs

```diff
@@ -49,6 +43,29 @@ impl Registry {
 
         Ok(rewards)
     }
+
+    /// Return the node rewards table from the registry.
+    pub fn get_node_rewards_table(&self) -> Result<NodeRewardsTable, String> {
+        let rewards_table_bytes = self
+            .get(NODE_REWARDS_TABLE_KEY.as_bytes(), self.latest_version())
+            .ok_or_else(|| "Node Rewards Table was not found in the Registry".to_string())?
+            .value
+            .clone();
+
+        Ok(NodeRewardsTable::decode(rewards_table_bytes.as_slice()).unwrap())
+    }
+
+    /// Check if the Node Rewards Table is empty.
+    pub fn is_node_rewards_table_empty(&self) -> bool {
+        if self
+            .get(NODE_REWARDS_TABLE_KEY.as_bytes(), self.latest_version())
+            .is_none()
+        {
+            return true;
+        }
+        let rewards_table_bytes = self.get_node_rewards_table().unwrap();
```

Comment:
This code would be a bit cleaner if we returned a typed error instead of a String "Node Rewards Table was not found in the Registry" -- we could then have a single match self.get_node_rewards_table() here. Should we introduce this typed error, or do we just leave it as is @max-dfinity?